chromium_src/url/url_canon_stdurl.cc

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "url/url_canon.h"
+
+namespace ipfs {
+template <typename CHAR>
+bool IpfsCIDv0(const url::URLComponentSource<CHAR>& source,
+               const url::Parsed& parsed,
+               url::CanonOutput* output,
+               url::Parsed* new_parsed);
+}  // namespace ipfs
+
+#define CanonicalizeHost \
+  ipfs::IpfsCIDv0<CHAR>(source, parsed, output, new_parsed) || CanonicalizeHost
+
+#include "../../url/url_canon_stdurl.cc"
+#undef CanonicalizeHost
+
+namespace ipfs {
+template <typename CHAR>
+// Do not canonicalize CIDv0(Qm...) and copy it as is

```

Comment:
Qm? Can you link to an ipfs doc explaining this in the comment?

Reply:
https://docs.ipfs.io/concepts/content-addressing/#version-0-v0

Reply:
added it to comment